cohortextractor/tpp_backend.py

```diff
@@ -3491,6 +3491,45 @@ def patients_with_an_ons_cis_record(
             """
         return table_queries + [sql]
 
+    def patients_with_record_in_ukrr(
+        self,
+        # picks dataset held by UKRR
+        from_dataset,
+        returning,
+        # Date filtering: date limits
+        on_or_before=None,
+        on_or_after=None,
+        between=None,
+        include_date_of_match=False,
+        date_format=None,
+        return_expectations=None,
+    ):
+        dataset_mapping = {"2019_prevalence": "2019prev", "2020_prevalence": "2020prev", "2021_prevalence": "2021prev",
+                           "2020_incidence": "2020inc", "2020_ckd": "2020ckd"}
+
+        date_condition, date_joins = self.get_date_condition(
+            "UKRR", "rrt_start", between
+        )
+
+        if returning == "binary_flag":
+            column_definition = "1"
+        elif returning == "treatment_modality_start":
+            column_definition = "mod_start"
+        elif returning == "treatment_modality_prevalence":
+            column_definition = "mod_prev"
+        elif returning == "latest_creatinine":
+            column_definition = "creat"
+        elif returning == "latest_egfr":
+            column_definition = "eGFR_ckdepi"
+        else:
+            column_definition = returning
```

Comment:
This could be nicer as another mapping:

```
column_mapping = {
    "binary_flag": 1,
    "treatment_modality_start": "mod_start",
    ...
}
column_definition = column_mapping.get(returning)
```

And then we can also check and raise an exception if the `returning` value isn't one we expected